### PR TITLE
OADP-2130: rework of the introduction section

### PR DESCRIPTION
**Pull request by anarnold97**
[OADP-2130](https://issues.redhat.com/browse/OADP-2130)

Doc Improvements - Restructure of OADP documentation

This PR is a focused on including an introduction section and improving the structure of the documentation as the taxonomy was problematic. 

Version(s):

Enterprise 4.11 → branch/enterprise-4.11
Enterprise 4.12 → branch/enterprise-4.12
Enterprise 4.13 → branch/enterprise-4.13
Enterprise 4.14 → branch/enterprise-4.14

Issue:

https://issues.redhat.com/browse/OADP-2130

Link to docs preview:

[OADP - Introduction](https://62396--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-intro.html)
QE review:
- [ ] QE has approved this change.

Additional information:

I am breaking the rework up into smaller chunks, so it will be easier for me to get them through OCP review. 
I hope this is ok, but the other options would be too difficult to manage.

### CODE REVIEW

**Comment by anarnold97** [comment]
@weshayutin @akarol 
Please can you or your respective teams review, so I can move to OCP review? Thanks

**Comment by amastbau** [comment]
Hi @anarnold97 
can you please paste a link directly to the preview?

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:13]: might be confusing
vm is also the short-hand name of the VirtualMachine Resouce type 
When br VM, it is not only the vm resource

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:19]: there are more velero resource types 
but I assume we include only the customer-facing ones, correct?

backuprepositories
backups
backupstoragelocations
deletebackuprequests
downloadrequests
podvolumebackups
podvolumerestores
restores
schedules
serverstatusrequests
volumesnapshotlocations

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:27]: was this missed?

**Review by amastbau — CHANGES_REQUESTED** [review]

**Comment by anarnold97** [comment]
> Hi @anarnold97 can you please paste a link directly to the preview?

Yeah, but I did, please see the top of the document where it say *Link to docs preview:* underneath you will see a link called *OADP - Introduction*

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:27]: No. It is a note for myself

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:19]: Yes

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:13]: but it says *virtual machine* in front of it so there is no confusion

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:27]: Please remember, that this restructure is split into three parts. Once each is changed, I need to come back and add links to these sections. 

However, until I have the sections approved I cannot link cleanly.

**Comment by amastbau** [comment]
@anarnold97 
for some reason
https://62396--docspreview.netlify.app/
takes me to the main page 
OpenShift Docs PR Preview
maybe I am missing something

**Comment by anarnold97** [comment]
> @anarnold97 for some reason https://62396--docspreview.netlify.app/ takes me to the main page OpenShift Docs PR Preview maybe I am missing something

@amastbau - as I said previously **Yeah, but I did, please see the top of the document where it say Link to docs preview: underneath you will see a link called OADP - Introduction**. 

image

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (amastbau) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:27]: @anarnold97 
This is still WIP, correct or is it ready?

Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:27]: It is mainly finished - I was trying to get the restructure done as there are so many links. Then do the re-write... but it looks like QE has decided I am doing everything at once. LOL

**Comment by anarnold97** [comment]
> @anarnold97 , @akarol
> 
> Hi, I don't think the bugs fixed list should be part of the release notes

Yeah, I hear you but standard release notes often include Resolved Issues. But you are right to some extent, I would prefer a Jira filter. 

I hate the formatting of these release notes and want to rework them as they look bad.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (snarayan-redhat) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:3]:
```suggestion
= Introduction to {oadp-full}
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (snarayan-redhat) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:11]:
```suggestion
The {oadp-first} product safeguards customer applications on the {product-title}. OADP provides disaster recovery protection for {product-title} applications, application-related cluster resources, persistent volumes, and internal images.
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (snarayan-redhat) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:15]:
```suggestion
OADP is not a disaster recovery solution for *etcd* or {product-title} Operators. For more information on etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].
```
OR
```suggestion
OADP cannot be a disaster recovery solution for *etcd* or OpenShift Operators. For more information on etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (snarayan-redhat) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]: I am not aware of the technicalities. But are those approaches = the following list of APIs?
If so, my suggestion:
```suggestion
OAP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.
```
else
 you might have to explain those multiple approaches.

File: _topic_maps/_topic_map.yml
Comment (snarayan-redhat) [inline _topic_maps/_topic_map.yml:2739]:
```suggestion
  - Name: Introduction to OpenShift API for Data Protection
```

**Review by snarayan-redhat — COMMENTED** [review]
Nice work. Minor nits and few suggestions!

Also, a reminder to get approval from other stakeholders before requesting merge.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]:
```suggestion
OADP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc]: (giggle) @snarayan-redhat - you spelt OADP wrong! What a bad reviewer! LOL

Seriously, thanks for the great help you gave with your review.

**Comment by anarnold97** [comment]
> Nice work. Minor nits and few suggestions!
> 
> Also, a reminder to get approval from other stakeholders before requesting merge.

Oh dear, had so many approvals!!!!

**Comment by stevsmit** [comment]
@anarnold97 A few suggestions before merging.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]: Please remember this is part of a re-work and once done, I need to come back and add links, which will hopefully clarify

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (michaelryanpeter) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:4]: In order to have the `{oadp-full}` variable render in the heading on the Customer Portal, you need to move the `include::` statement above the heading. You can see an example in the [contributor guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata).

This quirk of the Customer Portal is something we only found out about recently. It was brought up at the OpenShift docs weekly team meeting today, 21 August 2023.
```suggestion
[id="oadp-introduction"]
include::_attributes/common-attributes.adoc[]
= Introduction to {oadp-full}
```

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (michaelryanpeter) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]: +1 to Steven's comment. I am not sure that adding links will clarify the sentence. I find the infinitive phrases difficult to follow. Is this sentence is describing one, very specific action that these APIs enable, or if it is a list of capabilities that these APIs share? Do the APIs use namespaces and labels to avoid backing up the wrong resources? 

I understand that some of this content is a placeholder for future revisions.

**Review by michaelryanpeter — COMMENTED** [review]
There is one comment that must be addressed so that the `oadp-full` variable renders in the heading on the customer portal. I second Steven's comment about the lead-in sentence to the API list being difficult to follow. 

Lastly, I did not see an approval or LGTM from QE in the Jira or in the PR comments. Would you please verify that this has been ack'ed by QE?

Thanks!
/remove-label merge-review-needed
/remove-label merge-review-in-progress

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]: @michaelryanpeter - I am sorry not repeating myself

**Comment by anarnold97** [comment]
@kalexand-rh - this rework was signed-off as part of a larger rework - https://docs.google.com/document/d/1WRFepkp8ZmUKXJNi0Cbrho8_mVudpU710fHGtRgenCI/edit

This rework is taking place because the OADP docs are frankly appalling, and should never have been published as they stink, and we needed to do something major. 

As you can see the initial effort was to have a proper introduction, I will be adding links and better titles to mark the division between OADP backup and other backups such as Control Plane. But I cannot add links to other sections in the doc-set until I have those pages available. I am trying to address problems such as having customers who were reading the OADP docs and had no idea what OADP did.

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (kalexand-rh) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]: @anarnold97, I agree with Michael and Steven. I think you should comment out lines 17-26 until you can link to the API information. In this state, it's not providing additional value.

Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:17]: I originally had it separated, but the first reviewer wanted a change and suggested adding that sentence. So can I go back to something like the original:
```
There exists numerous approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels. 

OADP provides the following APIs:

    Backup

    Restore

    Schedule

    BackupStorageLocation

    VolumeSnapshotLocation
```
These would then link to the new main sections in the document.

**Comment by anarnold97** [comment]
https://github.com/openshift/openshift-docs/pull/61927

image

@akarol - please can you give me a LGTM so I can get this moving? Thanks

/label merge-review-needed

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (mburke5678) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:25]: @anarnold97 I am assuming here that the link for  `BackupSnapshotLocation` is the same as for `VolumeSnapshotLocation`. 
I am not sure which option would be less confusing, leaving `VolumeSnapshotLocation` without a link, or provide the same link for both. I defer to your guidance.

Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/oadp-intro.adoc:25]: Thanks @mburke5678 - will make this change later this evening

File: backup_and_restore/application_backup_and_restore/oadp-intro.adoc
Comment (stillalearner) [inline backup_and_restore/application_backup_and_restore/oadp-intro.adoc:23]: this link should redirect to the Scheduling Backups section.
https://62396--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications#oadp-scheduling-backups_backing-up-applications

I dont see anything related to Schedule on the current link it redirects to.

**Pull request closed without merging**
